src/Models/Substrate/OfferDataParams.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Enjin\Platform\Marketplace\Models\Substrate;
+
+class OfferDataParams
+{
+    /**
+     * Create a new instance of the model.
+     */
+    public function __construct(
+        public ?int $expiration = null,
+    ) {}
+
+    /**
+     * Convert the object to encodable formatted array.
+     */
+    public function toEncodable(): array
+    {
+        return ['startBlock' => $this->startBlock, 'endBlock' => $this->endBlock];
```

Comment:
Just checking, is this the right encodable params for the offer type?  Or should we be encoding the expiration?

Reply:
Thanks, I will check later today.